ee/katc/config.go

```diff
@@ -101,35 +101,57 @@ func (r *rowTransformStep) UnmarshalJSON(data []byte) error {
 // katcTableConfig is the configuration for a specific KATC table. The control server
 // sends down these configurations.
 type katcTableConfig struct {
+	Name              string             `json:"name"`
 	SourceType        katcSourceType     `json:"source_type"`
-	Source            string             `json:"source"` // Describes how to connect to source (e.g. path to db) -- % and _ wildcards supported
-	Platform          string             `json:"platform"`
+	SourcePaths       []string           `json:"source_paths"` // Describes how to connect to source (e.g. path to db) -- % and _ wildcards supported
+	Filter            string             `json:"filter"`
 	Columns           []string           `json:"columns"`
-	Query             string             `json:"query"` // Query to run against `path`
+	SourceQuery       string             `json:"source_query"` // Query to run against each source path
 	RowTransformSteps []rowTransformStep `json:"row_transform_steps"`
 }
 
 // ConstructKATCTables takes stored configuration of KATC tables, parses the configuration,
 // and returns the constructed tables.
 func ConstructKATCTables(config map[string]string, slogger *slog.Logger) []osquery.OsqueryPlugin {
 	plugins := make([]osquery.OsqueryPlugin, 0)
-	for tableName, tableConfigStr := range config {
+
+	tableConfigs, tableConfigsExist := config["tables"]
+	if !tableConfigsExist {
+		slogger.Log(context.TODO(), slog.LevelWarn,
+			"missing top-level tables key in KATC config, cannot construct tables",
+		)
+
+		return plugins
+	}
+
+	// We want to unmarshal each table config separately, so that we don't fail to configure all tables
+	// if only some are malformed.
+	var rawTableConfigs []json.RawMessage
+	if err := json.Unmarshal([]byte(tableConfigs), &rawTableConfigs); err != nil {
+		slogger.Log(context.TODO(), slog.LevelWarn,
+			"could not unmarshal tables in KATC config",
+			"err", err,
+		)
+		return plugins
+	}
+
+	for _, rawTableConfig := range rawTableConfigs {
 		var cfg katcTableConfig
-		if err := json.Unmarshal([]byte(tableConfigStr), &cfg); err != nil {
+		if err := json.Unmarshal(rawTableConfig, &cfg); err != nil {
 			slogger.Log(context.TODO(), slog.LevelWarn,
-				"unable to unmarshal config for Kolide ATC table, skipping",
-				"table_name", tableName,
+				"unable to unmarshal config for KATC table, skipping",
 				"err", err,
 			)
 			continue
 		}
 
-		if cfg.Platform != runtime.GOOS {
+		// For now, the filter is simply the OS
```

Comment:
What about making `filter` into `filters`, a map[string]string? Simple to unmarshal and therefore parse, and flexible for future additions?

```json
{
						"name": "kolide_indexeddb_leveldb_test",
						"source_type": "indexeddb_leveldb",
						"filters": {
						     "goos": "darwin"
						},
						"columns": ["data"],
						"source_paths": ["/some/path/to/db.indexeddb.leveldb"],
						"source_query": "db.store",
						"row_transform_steps": ["deserialize_chrome"]
					}
```

Reply:
`filters` makes sense to me. I don't know if that should be an array of strings, or a map. Array might be little more flexible, I think either can work. 

I think we should ponder whether that's an `AND` or an `OR`. If we go the way of overlays, this should be an `AND`, because the overlays handle `OR`

Reply:
I've updated to `filters` in the other PR https://github.com/kolide/launcher/pull/1772 -- if we decide to go with this approach I'll make the update here as well.